warehouse/ingest-core/src/main/java/datawave/ingest/config/BaseHdfsFileCacheUtil.java

```diff
@@ -34,15 +34,22 @@ public Path getCacheFilePath() {
     public abstract void setCacheFilePath(Configuration conf);
     
     public void read() throws IOException {
-        try (BufferedReader in = new BufferedReader(new InputStreamReader(FileSystem.get(this.cacheFilePath.toUri(), conf).open(this.cacheFilePath)))) {
-            readCache(in, delimiter);
-        } catch (IOException ex) {
-            if (shouldRefreshCache(this.conf)) {
-                update();
-            } else {
-                throw new IOException("Unable to read cache file at " + this.cacheFilePath, ex);
-            }
+        int attempts = 0;
+        boolean retry = true;
+        while (retry && attempts <= MAX_RETRIES) {
```

Comment:
no need to check for attempts here as you blow out with an exception when reaching this limit below.